org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/commands/OrganizeImportsCommandTest.java

```diff
@@ -234,6 +237,51 @@ public void testOrganizeImportsAutomaticallyResolve() throws CoreException, BadL
 		assertEquals(buf.toString(), getOrganizeImportResult(cu, rootEdit));
 	}
 
+	@Test
+	public void testOrganizeImportsFilterTypes() throws CoreException, BadLocationException {
+		List<String> filteredTypes = new ArrayList<>();
+		filteredTypes.add("java.util.*");
+		PreferenceManager.getPrefs(null).setFilteredTypes(filteredTypes);
+		try {
+			IPackageFragment pack1 = fSourceFolder.createPackageFragment("test1", false, null);
+
+			StringBuilder buf = new StringBuilder();
+			buf.append("package test1;\n");
+			buf.append("\n");
+			buf.append("public class E {\n");
+			buf.append("\n");
+			buf.append("    public E() {\n");
+			buf.append("        List list = new ArrayList();\n");
+			buf.append("    }\n");
+			buf.append("}\n");
+
+			ICompilationUnit cu = pack1.createCompilationUnit("E.java", buf.toString(), false, null);
+
+			buf = new StringBuilder();
+			buf.append("package test1;\n");
+			buf.append("\n");
+			buf.append("import java.util.ArrayList;\n");
+			buf.append("import java.util.List;\n");
+			buf.append("\n");
+			buf.append("public class E {\n");
+			buf.append("\n");
+			buf.append("    public E() {\n");
+			buf.append("        List list = new ArrayList();\n");
+			buf.append("    }\n");
+			buf.append("}\n");
+
+			WorkspaceEdit rootEdit = new WorkspaceEdit();
+			command.organizeImportsInCompilationUnit(cu, rootEdit);
+			assertEquals(0, rootEdit.getChanges().size());
+			PreferenceManager.getPrefs(null).setFilteredTypes(Collections.emptyList());
+			command.organizeImportsInCompilationUnit(cu, rootEdit);
+			assertFalse(rootEdit.getChanges().isEmpty());
+			assertEquals(buf.toString(), getOrganizeImportResult(cu, rootEdit));

```

Comment:
if you set an empty list for filtered types, how come the organize imports action managed to select java.util.List over java.awt.List?

Reply:
TypeFilter is never called in this test. I don't understand how it even passes

Reply:
> if you set an empty list for filtered types, how come the organize imports action managed to select java.util.List over java.awt.List?

java.awt.List doesn't when running tests.

> TypeFilter is never called in this test. I don't understand how it even passes

See org.eclipse.jdt.core.manipulation.TypeNameMatchCollector.getStringMatchers()

Reply:
Ok I suspect the fake rt jars we use don't have the java.awt classes.
Thanks for the info about TypeNameMatchCollector